tests/onnxruntime/test_modeling.py

```diff
@@ -61,6 +61,31 @@ def test_load_model_from_hub(self):
         model = ORTModel.from_pretrained(self.ONNX_MODEL_ID)
         self.assertIsInstance(model.model, onnxruntime.capi.onnxruntime_inference_collection.InferenceSession)
         self.assertIsInstance(model.config, PretrainedConfig)
+        self.assertListEqual(model.providers, ["CPUExecutionProvider"])
+        self.assertEqual(model.device, torch.device("cpu"))
+
+    def test_load_model_cuda_provider_list(self):
+        model = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider=["CUDAExecutionProvider"])
+        self.assertListEqual(model.providers, ["CUDAExecutionProvider", "CPUExecutionProvider"])
+        self.assertEqual(model.device, torch.device("cuda"))
+
+    def test_load_model_cuda_provider_str(self):
+        model = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider="CUDAExecutionProvider")
+        self.assertListEqual(model.providers, ["CUDAExecutionProvider", "CPUExecutionProvider"])
+        self.assertEqual(model.device, torch.device("cuda"))
+
+    def test_load_model_several_providers(self):
+        model = ORTModel.from_pretrained(
+            self.ONNX_MODEL_ID, provider=["CUDAExecutionProvider", "TensorrtExecutionProvider"]
```

Comment:
does it make sense? What happens if the model hast cuda and tensorrt?

Reply:
An error will be raised here: https://github.com/fxmarty/optimum/blob/21f4d919b202561968258fb6e8dbce67130e957b/optimum/onnxruntime/modeling_ort.py#L163-L167